### PR TITLE
feat: add `Logger` effect

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/library/Logger.flix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    pub def log(s: Severity, m: String): Unit
+
+}
+
+mod Logger {
+
+    ///
+    /// Logs the message `m` at the `Trace` level.
+    ///
+    pub def trace(m: a): Unit \ Logger with ToString[a] = do Logger.log(Severity.Trace, "${m}")
```
Comment (mlutze) [inline main/src/library/Logger.flix:34]: Is this interface typical? I would expect only to debug strings.

File: main/src/library/Logger.flix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    ///
+    /// Logs the message `m` at the `Debug` level.
+    ///
+    pub def debug1(m: a): Unit \ Logger with ToString[a] = do Logger.log(Severity.Debug, "${m}")
```
Comment (mlutze) [inline main/src/library/Logger.flix:39]: I guess there is a TODO here

**Review by mlutze — APPROVED** [review]

File: main/src/library/Logger.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/src/library/Logger.flix:34]: No strong opinion, but this way you can debug a number too

File: main/src/library/Logger.flix (hunk 2 above)
Comment (magnus-madsen) [reply main/src/library/Logger.flix:39]: Yes, with ticket: https://github.com/flix/flix/issues/8626

**Pull request merged**